lib/formatters/tapFormatter.mjs

```diff
@@ -1,46 +1,53 @@
+import { isNumber } from '../utils/validateTypes.mjs';
 import preprocessWarnings from './preprocessWarnings.mjs';
 
-/**
- * @type {import('stylelint').Formatter}
- */
+/** @typedef {import('stylelint').Warning} Warning */
+/** @typedef {Array<Omit<Warning, 'rule'>>} Warnings */
+
+/** @type {import('stylelint').Formatter} */
 export default function tapFormatter(results) {
-	const lines = [`TAP version 13\n1..${results.length}`];
+	const lines = [`TAP version 14\n1..${results.length}`];
 
 	for (const [index, result] of results.entries()) {
 		preprocessWarnings(result);
-
-		lines.push(
-			`${result.errored ? 'not ok' : 'ok'} ${index + 1} - ${result.ignored ? 'ignored ' : ''}${
-				result.source
-			}`,
-		);
-
-		if (result.warnings.length > 0) {
-			lines.push('---', 'messages:');
-
-			for (const warning of result.warnings) {
-				lines.push(
-					` - message: "${warning.text}"`,
-					`   severity: ${warning.severity}`,
-					`   data:`,
-					`     line: ${warning.line}`,
-					`     column: ${warning.column}`,
-				);
-
-				if (typeof warning.endLine === 'number') {
-					lines.push(`     endLine: ${warning.endLine}`);
-				}
-
-				if (typeof warning.endColumn === 'number') {
-					lines.push(`     endColumn: ${warning.endColumn}`);
-				}
-
-				if (typeof warning.rule === 'string') {
-					lines.push(`     ruleId: ${warning.rule}`);
-				}
+		// SKIP indicates "that a test was not run, or if it was,
+		// that its success or failure is being temporarily ignored."
+		// skipped tests must not be treated as test failures though
+		// if a file is ignored, errored is currently always false
+		const ignored = result.ignored ? ' # SKIP ignored' : '';
```

Comment:
[ask] Is this `ignored` word at the end necessary? I cannot find the word with `# SKIP` in the TAP14 document.

```suggestion
		const ignored = result.ignored ? ' # SKIP' : '';
```

Reply:
`ignored` is the "skip reason"